src/v/cluster/members_backend.cc

```diff
@@ -226,6 +226,36 @@ ss::future<> members_backend::calculate_reallocations(update_meta& meta) {
         co_return;
     }
 }
+
+void members_backend::reallocations_for_even_partition_count(
+  members_backend::update_meta& meta, partition_allocation_domain domain) {
+    calculate_reallocations_for_even_partition_count(meta, domain);
+
+    auto current_error = calculate_unevenness_error(domain);
+    auto [it, _] = meta.last_unevenness_error.try_emplace(domain, 1.0);
+    const auto min_improvement = std::max<size_t>(
```

Comment:
I wonder if it would be simpler if we just calculated the number of moves that result in improvement? Or is this different?

Reply:
i think it is different and similar at the same time :). we do not have information for particular reassignment hence we calculate the step improvement and based on that calculate how many reassignments cause improvements.